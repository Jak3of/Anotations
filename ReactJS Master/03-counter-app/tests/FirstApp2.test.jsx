import { describe, test , expect } from '@jest/globals';

import { render , screen } from '@testing-library/react';
import FirstApp  from '../src/FisrtApp';

describe('Pr_Test en <FisrtApp/>',() => {
    
    const title = 'Hola soy Vegeta';
    const subTitle = 'insecto';
    
    test('debe de hacer match con el snapshot',() =>{
    
        const {container} = render(<FirstApp title={title} subTitle={subTitle} />);

        expect(container).toMatchSnapshot();

        

        

    })

    test('debe de mostrar el titulo',() =>{
            
        render(<FirstApp title={title}  />);
        // lee todo el html en <body></body>
        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();

    })

    test('debe de mostrar el titulo en un h1',() =>{

        render(<FirstApp title={title}  />);
        expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain( title );
        
    })

    test('debe de mostrar el subtitulo enviado por props', () => { 

        render(<FirstApp title={title}  subTitle={subTitle}  />);
        expect(screen.getAllByText(subTitle).length).toBe(1);

     })


    test('debe de mostrar el subtitulo dentro de algun p', () => {
        
        render(<FirstApp title={title}  subTitle={subTitle}  />);

        expect(screen.getByText(subTitle)).toBeTruthy();
    })
})