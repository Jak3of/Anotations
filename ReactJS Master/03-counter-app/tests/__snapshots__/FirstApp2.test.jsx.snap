// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pr_Test en <FisrtApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy Vegeta
  </h1>
  <p>
    insecto
  </p>
  <p>
    Nelson
  </p>
</div>
`;
